File: Java/Length-of-Last-Word.java
public class Solution {
    /**
     * @param s A string
     * @return the length of last word
     */
    public int lengthOfLastWord(String s) {
        // Write your code here
        int cur = s.length() - 1;
        while (cur >= 0 && s.charAt(cur) == ' ') cur --;
        int res = 0;
        while (cur >= 0 && s.charAt(cur) != ' ') {
        	res ++;
        	cur --;
        }
        return res;
    }
}

File: Python/Length-of-Last-Word.py
class Solution:
    # @param {string} s A string
    # @return {int} the length of last word
    def lengthOfLastWord(self, s):
        # Write your code here
        p = len(s) - 1
        while p >= 0 and s[p] == ' ':
            p -= 1
        if p < 0:
            return 0
        cur = 0
        while p >= 0 and s[p] != ' ':
            cur += 1
            p -= 1
        return cur